primitives/application-crypto/test/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Integration tests for application crypto
+
+
+#[cfg(test)]
+mod ed25519 {
```

Comment:
Common ben, both of these modules can be removed and the tests can live in the crate.

Reply:
:P

Reply:
I went for minimal changes via copy-paste. Don't have much of an opinion other than that feels like appropriate compartmentalizing inline with what the original crate did. Consider that nothing here will ever be imported, either is fine with me. If you insist I will flatten it.

Reply:
Either flatten or what @tomusdrw says.